### PR TITLE
Add a skeleton of the sunset (deprecation) mechanism.

### PR DESCRIPTION
**Pull request by msramek**
Added a new module with the deprecation code.

What's missing yet:
- notification texts
- linked webpages
- any errback handling

### CODE REVIEW

File: chrome/sunset.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
+// Copyright 2011 Google Inc.
```
Comment (mikewest) [inline chrome/sunset.js]: 2015. :)

File: chrome/sunset.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    "date": new Date().toString()
+  });
+}
+
+
+/**
+ * Triggers a loop that tests if conditions were met to show the message.
+ */
+Sunset.run = function() {
+  setInterval(
```
Comment (mikewest) [inline chrome/sunset.js]: https://developer.chrome.com/apps/alarms is probably what you want here.

File: chrome/sunset.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+Sunset.timeline_offsets_ = [
+  7,
+  14,
+  28,
+  29
+];
+
+
+/**
+ * The minimum offset ind days between two consequent notifications.
```
Comment (mikewest) [inline chrome/sunset.js]: Nit: s/ind/in/

Comment (mikewest) [inline chrome/sunset.js]: Nit: s/consequent/subsequent/

File: chrome/sunset.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+ */
+Sunset.maybeShowNotification_ = function() {
+  chrome.storage.sync.get(null, function(data) {
+    // Read the index of the notification to be shown and the date when
+    // the previous notification was shown.
+    var index = parseInt(data.index, 10);
+    var last_date = new Date(data.date);
+
+    // Reset the index if the stored data are invalid or not present.
+    if (isNaN(index) || index < 0 || index > 3 || !last_date.getTime()) {
```
Comment (mikewest) [inline chrome/sunset.js]: Nit: s/3/Sunset.timeline_offsets_.length/

File: chrome/sunset.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+      index = 0;
+    }
```
Comment (mikewest) [inline chrome/sunset.js]: Nit: No {} for one line clauses.

File: chrome/sunset.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+/**
+ * Shows a notification.
+ * @private
+ */
+Sunset.showNotification_ = function(index) {
+  // There is no 4th notification. Instead, we just uninstall the extension.
+  if (index == 3) {
+    chrome.management.uninstallSelf();
+  }
```
Comment (mikewest) [inline chrome/sunset.js]: Nit: No {}.

File: chrome/sunset.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+};
+
+
+/**
+ * Shows a notification.
+ * @private
+ */
+Sunset.showNotification_ = function(index) {
+  // There is no 4th notification. Instead, we just uninstall the extension.
+  if (index == 3) {
```
Comment (mikewest) [inline chrome/sunset.js]: Nit: >= is safer.

File: chrome/sunset.js [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+    timeline_threshold.setDate(
+        timeline_threshold.getDate() + Sunset.timeline_offsets_[index]);
+
+    var offset_threshold = last_date;
+    offset_threshold.setDate(
+        offset_threshold.getDate() + Sunset.minimum_offset_);
+
+    if (!index || (today >= timeline_threshold && today >= offset_threshold)) {
+      Sunset.showNotification_(index);
+    }
```
Comment (mikewest) [inline chrome/sunset.js]: Nit: No {}.

**Comment by mikewest** [comment]
There's a first set of comments. If you'd like to land this and iterate on it, LGTM.

File: chrome/sunset.js (hunk 5 above)
Comment (msramek) [inline chrome/sunset.js]: The condition expanded to two lines, so I kept the braces here.

**Comment by mikewest** [comment]
Land it whenever you like.

**Pull request merged**